src/psyclone/psyir/nodes/acc_directives.py

```diff
@@ -101,6 +101,36 @@ class ACCStandaloneDirective(ACCDirective, StandaloneDirective):
     ''' Base class for all standalone OpenACC directive statements. '''
 
 
+class ACCRoutineDirective(ACCStandaloneDirective):
+    ''' Class representing a "!$ACC routine" OpenACC directive in PSyIR. '''
+
+    def gen_code(self, parent):
+        '''Generate the fortran ACC Routine Directive and any associated
+        code.
+
+        :param parent: the parent Node in the Schedule to which to add our \
+                       content.
+        :type parent: sub-class of :py:class:`psyclone.f2pygen.BaseGen`
+        '''
+        # Check the constraints are correct
+        self.validate_global_constraints()
+
+        # Generate the code for this Directive
+        parent.add(DirectiveGen(parent, "acc", "begin", "routine", ""))
+
+    def begin_string(self):
```

Comment:
Could we replace this with the new Clause support now?

Reply:
Ideally I think we may want to support clauses such as `seq` and `vector` but I'll check with the various project members as to whether we have an immediate requirement.

Reply:
I have not added clauses yet as it seems al current cases will do with the standard directive.

That is if I understand you correctly, when adding clauses we will still use the begin_string as it is now. It will be the backend that will concatenate the begin_string with the result of visiting the children clauses.